packages/profiling-node/src/cpu_profiler.ts

```diff
@@ -15,6 +15,12 @@ import type {
 } from './types';
 import type { ProfileFormat } from './types';
 
+// #START_SENTRY_ESM_SHIM
+// When building for ESM, we shim require to use createRequire and __dirname.
+// We need to do this because .node extensions in esm are not supported.
```

Comment:
can you double check what happens when you do Node 22 and run node with `--experimental-require-module` (as per https://joyeecheung.github.io/blog/2024/03/18/require-esm-in-node-js/)

Reply:
Let me add it to the test (local is annoying because build fails as the binaries are missing)